scripts/scil_compute_divide.py

```diff
@@ -0,0 +1,226 @@
+    assert_outputs_exist(parser, args, arglist)
+
+    if not (len(args.in_dwis) == len(args.in_bvals)
+            == len(args.in_bvecs) == len(args.in_bdeltas)):
+        msg = """The number of given dwis, bvals, bvecs and bdeltas must be the
+              same. Please verify that all inputs were correctly inserted."""
+        raise ValueError(msg)
+
+    if len(np.unique(args.in_bdeltas)) < 2:
+        msg = """At least two different b-tensor shapes are needed for the
+              script to work properly."""
+        raise ValueError(msg)
+
+    affine = extract_affine(args.in_dwis)
+
+    tol = args.tolerance
+    force_b0_thr = args.force_b0_threshold
+
+    data, gtab_infos = generate_btensor_input(args.in_dwis,
+                                              args.in_bvals,
+                                              args.in_bvecs,
+                                              args.in_bdeltas,
+                                              force_b0_thr,
+                                              do_pa_signals=True,
+                                              tol=tol)
+
+    gtab_infos[0] *= 1e6  # getting bvalues to SI units
+
+    if args.mask is None:
+        mask = None
+    else:
+        mask = get_data_as_mask(nib.load(args.mask), dtype=bool)
+        if mask.shape != data.shape[:-1]:
+            raise ValueError("Mask is not the same shape as data.")
+
+    if args.fa is not None:
+        vol = nib.load(args.fa)
+        FA = vol.get_fdata(dtype=np.float32)
+
+    parameters = fit_gamma(data, gtab_infos, mask=mask,
+                           fit_iters=args.fit_iters,
+                           random_iters=args.random_iters,
+                           do_weight_bvals=args.do_weight_bvals,
+                           do_weight_pa=args.do_weight_pa,
+                           do_multiple_s0=args.do_multiple_s0,
+                           nbr_processes=args.nbr_processes)
+
+    microFA, MK_I, MK_A, MK_T = gamma_fit2metrics(parameters)
+    microFA[np.isnan(microFA)] = 0
+    microFA = np.clip(microFA, 0, 1)
+
+    if args.md:
+        nib.save(nib.Nifti1Image(parameters[..., 1].astype(np.float32),
+                                 affine), args.md)
+
+    if args.ufa:
+        nib.save(nib.Nifti1Image(microFA.astype(np.float32), affine), args.ufa)
+    if args.op:
+        if args.fa is not None:
+            OP = np.sqrt((3 * (microFA ** (-2)) - 2) / (3 * (FA ** (-2)) - 2))
+            OP[microFA < FA] = 0
+            nib.save(nib.Nifti1Image(OP.astype(np.float32), affine), args.op)
+        else:
+            logging.warning('The FA must be given in order to compute the OP.')
+    if args.mk_i:
+        nib.save(nib.Nifti1Image(MK_I.astype(np.float32), affine), args.mk_i)
+    if args.mk_a:
+        nib.save(nib.Nifti1Image(MK_A.astype(np.float32), affine), args.mk_a)
+    if args.mk_t:
+        nib.save(nib.Nifti1Image(MK_T.astype(np.float32), affine), args.mk_t)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
(CVXPY) Mar 16 03:22:46 PM: Encountered unexpected exception importing solver SCS:
ImportError('dlopen(/Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/_scs_direct.cpython-310-darwin.so, 0x0002): Symbol not found: ___emutls_get_address\n  Referenced from: <17C4D491-3B6D-344A-8328-189690282A3A> /Users/desm2239/my_env/scilpy_p10/lib/python3.10/site-packages/scs/.dylibs/libopenblasp-r0.3.21.dylib\n  Expected in:     <E0C76B2B-0EC6-364E-AACD-42C7AAE73F23> /Applications/freesurfer/lib/gcc/lib/libgomp.1.dylib')